public/app/plugins/datasource/influxdb/components/ConfigEditor.tsx

```diff
@@ -199,8 +200,8 @@ export class ConfigEditor extends PureComponent<Props, State> {
               <Input
                 id={`${htmlPrefix}-db`}
                 className="width-20"
-                value={options.database || ''}
-                onChange={onUpdateDatasourceOption(this.props, 'database')}
+                value={options.jsonData.dbName ?? options.database}
+                onChange={onUpdateDatasourceJsonDataOption(this.props, 'dbName')}
```

Comment:
i would recommend in the `onChange` also setting the `old database field` to empty-string